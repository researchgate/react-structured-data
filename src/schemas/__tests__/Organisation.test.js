import React from "react";
import renderer from "react-test-renderer";
import { ImageObject, JSONLD, Organization, Place } from "../../index";

describe("Organization example", () => {
  it("renders Organization with logo image", () => {
    expect(
      renderer.create(
        <JSONLD>
          <Organization
            type="publisher"
            url="https://example.org/institution/ResearchGate"
            name="ResearchGate"
          >
            <Place type="location" name="Berlin, Germany" />
            <ImageObject
              type="logo"
              url="https://example.org/ii/institution.image/image_url.jpeg"
            />
          </Organization>
        </JSONLD>
      )
    ).toMatchSnapshot();
  });
  it("renders Organization with department", () => {
    expect(
      renderer.create(
        <JSONLD>
          <Organization
            type="publisher"
            url="https://example.org/institution/ResearchGate"
            name="ResearchGate"
          >
            <Place type="location" name="Berlin, Germany" />
            <ImageObject
              type="logo"
              url="https://example.org/ii/institution.image/image_url.jpeg"
            />
            <Organization
              type="department"
              url="https://example.org/institution/ResearchGate/department/Engineering"
              name="Engineering"
            />
          </Organization>
        </JSONLD>
      )
    ).toMatchSnapshot();
  });
});
