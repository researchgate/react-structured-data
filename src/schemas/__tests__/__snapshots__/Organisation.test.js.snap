// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization example renders Organization with department 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"https://schema.org/","@type":"Organization","url":"https://example.org/institution/ResearchGate","name":"ResearchGate","location":{"@type":"Place","name":"Berlin, Germany"},"logo":{"@type":"ImageObject","url":"https://example.org/ii/institution.image/image_url.jpeg"},"department":{"@type":"Organization","url":"https://example.org/institution/ResearchGate/department/Engineering","name":"Engineering"}}
</script>
`;

exports[`Organization example renders Organization with logo image 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"https://schema.org/","@type":"Organization","url":"https://example.org/institution/ResearchGate","name":"ResearchGate","location":{"@type":"Place","name":"Berlin, Germany"},"logo":{"@type":"ImageObject","url":"https://example.org/ii/institution.image/image_url.jpeg"}}
</script>
`;
